types/module/server/manager.go

```diff
@@ -0,0 +1,147 @@
+package server
+
+import (
+	"fmt"
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkmodule "github.com/cosmos/cosmos-sdk/types/module"
+	gogogrpc "github.com/gogo/protobuf/grpc"
+	"github.com/regen-network/regen-ledger/types/module"
+	"reflect"
+)
+
+type Manager struct {
+	sdkmodule.Manager
+	baseApp          *baseapp.BaseApp
+	cdc              *codec.ProtoCodec
+	keys             map[string]ModuleKey
+	router           *router
+	requiredServices map[reflect.Type]bool
+}
+
+func NewManager(baseApp *baseapp.BaseApp, cdc *codec.ProtoCodec) *Manager {
+	return &Manager{
+		baseApp: baseApp,
+		cdc:     cdc,
+		keys:    map[string]ModuleKey{},
+		router: &router{
+			handlers:         map[string]handler{},
+			providedServices: map[reflect.Type]bool{},
+			antiReentryMap:   map[string]bool{},
+		},
+	}
+}
+
+func (mm *Manager) RegisterModules(modules []module.ModuleBase) error {
+	for _, mod := range modules {
+		// check if we actually have a server module, otherwise skip
+		serverMod, ok := mod.(Module)
+		if !ok {
+			continue
+		}
+
+		serverMod.RegisterInterfaces(mm.cdc.InterfaceRegistry())
+	}
+
+	for _, mod := range modules {
```

Comment:
you can use the same loop as before I think :)

Reply:
I want to make sure all interface types are registered first